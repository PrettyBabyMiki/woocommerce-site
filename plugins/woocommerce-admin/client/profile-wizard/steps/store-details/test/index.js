/**
 * External dependencies
 */
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

/**
 * Internal dependencies
 */
import { StoreDetails } from '../';

const testProps = {
	query: {
		page: 'wc-admin',
		path: '/setup-wizard',
	},
	step: {
		key: 'store-details',
		label: 'Store Details',
		isComplete: false,
	},
	initialValues: {
		addressLine1: '',
		addressLine2: '',
		city: '',
		countryState: '',
		postCode: '',
		isAgreeMarketing: true,
		storeEmail: 'wordpress@example.com',
	},
	getLocale: jest.fn(),
	isLoading: false,
	createNotice: jest.fn(),
	profileItems: {},
	updateAndPersistSettingsForGroup: jest.fn(),
	goToNextStep: jest.fn(),
	errorsRef: {
		current: {},
	},
};

jest.mock( '@wordpress/data', () => {
	const originalModule = jest.requireActual( '@wordpress/data' );

	return {
		__esModule: true,
		...originalModule,
		useSelect: jest.fn().mockReturnValue( {
			locale: 'en_US',
			countries: [
				{
					code: 'US',
					name: 'United States',
					states: [],
				},
			],
			loadingCountries: false,
			hasFinishedResolution: true,
		} ),
	};
} );

describe( 'StoreDetails', () => {
	describe( 'Snapshot test', () => {
		test( 'should match saved snapshot', () => {
			const container = render( <StoreDetails { ...testProps } /> );
			expect( container ).toMatchSnapshot();
		} );
	} );

	it( 'should disable the "Continue" button when the mandatory field (Country / Region) is empty', () => {
		const { getByRole } = render( <StoreDetails { ...testProps } /> );
		expect( getByRole( 'button', { name: 'Continue' } ) ).toBeDisabled();
	} );

	it( 'should enable the "Continue" button when the mandatory field (Country / Region) is filled', () => {
		const { getByRole } = render(
			<StoreDetails
				{ ...{
					...testProps,
					initialValues: {
						...testProps.initialValues,
						countryState: 'US',
					},
				} }
			/>
		);

		expect(
			getByRole( 'button', { name: 'Continue' } )
		).not.toBeDisabled();
	} );

	test( 'should call updateProfileItems when continue button is clicked given the allowTracking is true', async () => {
		const updateProfileItems = jest.fn();
		const { getByText } = render(
			<StoreDetails
				{ ...{
					...testProps,
					initialValues: {
						...testProps.initialValues,
						countryState: 'US',
					},
				} }
				updateProfileItems={ updateProfileItems }
				allowTracking={ true }
			/>
		);
		const continueButton = getByText( 'Continue' );
		userEvent.click( continueButton );
		await waitFor( () => {
			expect( updateProfileItems ).toHaveBeenCalled();
		} );
	} );

	describe( 'Email validation test cases', () => {
		test( 'should fail email validation and disable continue button when isAgreeMarketing is true and email is empty', async () => {
			const container = render(
				<StoreDetails
					{ ...testProps }
					initialValues={ {
						addressLine1: 'address1',
						addressLine2: 'address2',
						city: 'city',
						countryState: 'state',
						postCode: '123',
						isAgreeMarketing: true,
						storeEmail: 'wordpress@example.com',
					} }
				/>
			);
			const emailInput = container.getByLabelText( 'Email address' );
			await userEvent.clear( emailInput );
			userEvent.tab();
			expect(
				container.queryByText(
					'Please enter your email address to subscribe'
				)
			).toBeInTheDocument();

			expect(
				container.queryByRole( 'button', {
					name: 'Continue',
				} ).disabled
			).toBe( true );
		} );

		// test cases taken from wordpress php is_email test cases
		// https://github.com/WordPress/wordpress-develop/blob/2648a5f984b8abf06872151898e3a61d3458a628/tests/phpunit/tests/formatting/isEmail.php
		test.each( [
			'khaaaaaaaaaaaaaaan!',
			'http://bob.example.com/',
			"sif i'd give u it, spamer!1",
			'com.exampleNOSPAMbob',
			'bob@your mom',
			'a@b.c',
		] )( 'should fail email validation when given %s', async ( email ) => {
			const container = render( <StoreDetails { ...testProps } /> );
			const emailInput = container.getByLabelText( 'Email address' );
			await userEvent.clear( emailInput );
			await userEvent.type( emailInput, email );
			// validation is triggered onChange but error message only renders on blur
			// react testing lib doesn't have a "blur" event that can be triggered so this does the job of triggering the error message rendering
			userEvent.tab();
			expect(
				container.queryByText( 'Invalid email address' )
			).toBeInTheDocument();
		} );

		test.each( [
			'bob@example.com',
			'phil@example.info',
			// 'ace@204.32.222.14', this testcase passes for the backend validation but fails here, following up in a PR to fix this in @wordpress/url
			'kevin@many.subdomains.make.a.happy.man.edu',
			'a@b.co',
			'bill+ted@example.com',
		] )( 'should pass email validation when given %s', async ( email ) => {
			const container = render( <StoreDetails { ...testProps } /> );
			const emailInput = container.getByLabelText( 'Email address' );
			await userEvent.clear( emailInput );
			await userEvent.type( emailInput, email );
			userEvent.tab();
			expect(
				container.queryByText( 'Invalid email address' )
			).toBeNull();
		} );

		test( 'should pass email validation when field is empty', async () => {
			const container = render( <StoreDetails { ...testProps } /> );
			const emailInput = container.getByLabelText( 'Email address' );
			await userEvent.clear( emailInput );
			userEvent.tab();
			expect(
				container.queryByText( 'Invalid email address' )
			).toBeNull();
		} );
	} );
} );
