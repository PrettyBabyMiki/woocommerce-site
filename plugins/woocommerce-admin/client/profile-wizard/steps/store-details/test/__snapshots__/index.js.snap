// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreDetails Snapshot test should match saved snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="woocommerce-profile-wizard__store-details"
      >
        <div
          class="woocommerce-profile-wizard__step-header"
        >
          <h2
            class="components-truncate components-text css-7a2wz9-View-Text-sx-Base em57xhy0"
            data-wp-c16t="true"
            data-wp-component="Text"
          >
            Welcome to WooCommerce
          </h2>
          <p
            class="components-truncate components-text css-sgni2d-View-Text-sx-Base em57xhy0"
            data-wp-c16t="true"
            data-wp-component="Text"
          >
            Tell us where you run your business to help us configure currency, shipping, taxes, and more in a fully automated way.
          </p>
        </div>
        <div
          class="components-surface components-card css-nsno0f-View-Surface-getBorders-primary-Card-rounded em57xhy0"
          data-wp-c16t="true"
          data-wp-component="Card"
        >
          <div
            class="css-mgwsf4-View-Content em57xhy0"
          >
            <div
              class="components-card__body components-card-body css-1nwhnu3-View-Body-borderRadius-medium em57xhy0"
              data-wp-c16t="true"
              data-wp-component="CardBody"
            >
              <div
                class="woocommerce-store-address-fields"
              >
                <div>
                  <div
                    class="woocommerce-select-control is-searchable"
                  >
                    <input
                      autocomplete="country"
                      class="woocommerce-select-control__autofill-input"
                      name="country"
                      tabindex="-1"
                      type="text"
                      value=""
                    />
                    <input
                      autocomplete="address-level1"
                      class="woocommerce-select-control__autofill-input"
                      name="state"
                      tabindex="-1"
                      type="text"
                      value=""
                    />
                    <div
                      class="components-base-control woocommerce-select-control__control empty"
                    >
                      <svg
                        aria-hidden="true"
                        class="woocommerce-select-control__control-icon"
                        focusable="false"
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13.5 6C10.5 6 8 8.5 8 11.5c0 1.1.3 2.1.9 3l-3.4 3 1 1.1 3.4-2.9c1 .9 2.2 1.4 3.6 1.4 3 0 5.5-2.5 5.5-5.5C19 8.5 16.5 6 13.5 6zm0 9.5c-2.2 0-4-1.8-4-4s1.8-4 4-4 4 1.8 4 4-1.8 4-4 4z"
                        />
                      </svg>
                      <div
                        class="components-base-control__field"
                      >
                        <label
                          class="components-base-control__label"
                          for="woocommerce-select-control-0__control-input"
                        >
                          Country / Region *
                        </label>
                        <input
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-label="Country / Region *"
                          autocomplete="new-password"
                          class="woocommerce-select-control__control-input"
                          id="woocommerce-select-control-0__control-input"
                          placeholder=""
                          role="combobox"
                          type="search"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="components-base-control muriel-component muriel-input-text empty css-wdf2ti-Wrapper ej5x27r4"
                  >
                    <div
                      class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                    >
                      <label
                        class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                        for="inspector-text-control-0"
                      >
                        Address
                      </label>
                      <input
                        autocomplete="address-line1"
                        class="components-text-control__input"
                        id="inspector-text-control-0"
                        placeholder="Address"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="components-base-control muriel-component muriel-input-text empty css-wdf2ti-Wrapper ej5x27r4"
                  >
                    <div
                      class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                    >
                      <label
                        class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                        for="inspector-text-control-1"
                      >
                        Post code
                      </label>
                      <input
                        autocomplete="postal-code"
                        class="components-text-control__input"
                        id="inspector-text-control-1"
                        placeholder="Post code"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    class="components-base-control muriel-component muriel-input-text empty css-wdf2ti-Wrapper ej5x27r4"
                  >
                    <div
                      class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                    >
                      <label
                        class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                        for="inspector-text-control-2"
                      >
                        City
                      </label>
                      <input
                        autocomplete="address-level2"
                        class="components-text-control__input"
                        id="inspector-text-control-2"
                        placeholder="City"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="components-base-control muriel-component muriel-input-text with-value css-wdf2ti-Wrapper ej5x27r4"
                >
                  <div
                    class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                  >
                    <label
                      class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                      for="inspector-text-control-3"
                    >
                      Email address
                    </label>
                    <input
                      autocomplete="email"
                      checked=""
                      class="components-text-control__input"
                      id="inspector-text-control-3"
                      placeholder="Email address"
                      required=""
                      type="text"
                      value="wordpress@example.com"
                    />
                  </div>
                </div>
              </div>
              <div
                class="components-flex-item css-mw3lhz-View-Item-sx-Base em57xhy0"
                data-wp-c16t="true"
                data-wp-component="FlexItem"
              >
                <div
                  class="woocommerce-profile-wizard__newsletter-signup"
                >
                  <div
                    class="components-base-control components-checkbox-control css-wdf2ti-Wrapper ej5x27r4"
                  >
                    <div
                      class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                    >
                      <span
                        class="components-checkbox-control__input-container"
                      >
                        <input
                          checked=""
                          class="components-checkbox-control__input"
                          id="inspector-checkbox-control-0"
                          type="checkbox"
                          value="true"
                        />
                        <svg
                          aria-hidden="true"
                          class="components-checkbox-control__checked"
                          focusable="false"
                          height="24"
                          role="presentation"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
                          />
                        </svg>
                      </span>
                      <label
                        class="components-checkbox-control__label"
                        for="inspector-checkbox-control-0"
                      >
                        Get tips, product updates and inspiration straight to your mailbox.
                         
                        <span
                          class="woocommerce-profile-wizard__powered-by-mailchimp"
                        >
                          Powered by Mailchimp
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="components-flex components-card__footer components-card-footer css-142znvl-View-Flex-sx-Base-sx-Items-ItemsRow-Footer-borderRadius-borderColor-medium em57xhy0"
              data-wp-c16t="true"
              data-wp-component="CardFooter"
            >
              <button
                aria-disabled="true"
                class="components-button is-primary"
                disabled=""
                type="button"
              >
                Continue
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
            data-wp-c16t="true"
            data-wp-component="Elevation"
          />
          <div
            aria-hidden="true"
            class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
            data-wp-c16t="true"
            data-wp-component="Elevation"
          />
        </div>
        <div
          class="woocommerce-profile-wizard__footer"
        >
          <button
            class="components-button woocommerce-profile-wizard__footer-link is-link"
            type="button"
          >
            Skip setup store details
          </button>
          <button
            aria-label="Manual setup is only recommended for
 experienced WooCommerce users or developers."
            class="components-button is-tertiary"
            type="button"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3.2c-4.8 0-8.8 3.9-8.8 8.8 0 4.8 3.9 8.8 8.8 8.8 4.8 0 8.8-3.9 8.8-8.8 0-4.8-4-8.8-8.8-8.8zm0 16c-4 0-7.2-3.3-7.2-7.2C4.8 8 8 4.8 12 4.8s7.2 3.3 7.2 7.2c0 4-3.2 7.2-7.2 7.2zM11 17h2v-6h-2v6zm0-8h2V7h-2v2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="woocommerce-profile-wizard__store-details"
    >
      <div
        class="woocommerce-profile-wizard__step-header"
      >
        <h2
          class="components-truncate components-text css-7a2wz9-View-Text-sx-Base em57xhy0"
          data-wp-c16t="true"
          data-wp-component="Text"
        >
          Welcome to WooCommerce
        </h2>
        <p
          class="components-truncate components-text css-sgni2d-View-Text-sx-Base em57xhy0"
          data-wp-c16t="true"
          data-wp-component="Text"
        >
          Tell us where you run your business to help us configure currency, shipping, taxes, and more in a fully automated way.
        </p>
      </div>
      <div
        class="components-surface components-card css-nsno0f-View-Surface-getBorders-primary-Card-rounded em57xhy0"
        data-wp-c16t="true"
        data-wp-component="Card"
      >
        <div
          class="css-mgwsf4-View-Content em57xhy0"
        >
          <div
            class="components-card__body components-card-body css-1nwhnu3-View-Body-borderRadius-medium em57xhy0"
            data-wp-c16t="true"
            data-wp-component="CardBody"
          >
            <div
              class="woocommerce-store-address-fields"
            >
              <div>
                <div
                  class="woocommerce-select-control is-searchable"
                >
                  <input
                    autocomplete="country"
                    class="woocommerce-select-control__autofill-input"
                    name="country"
                    tabindex="-1"
                    type="text"
                    value=""
                  />
                  <input
                    autocomplete="address-level1"
                    class="woocommerce-select-control__autofill-input"
                    name="state"
                    tabindex="-1"
                    type="text"
                    value=""
                  />
                  <div
                    class="components-base-control woocommerce-select-control__control empty"
                  >
                    <svg
                      aria-hidden="true"
                      class="woocommerce-select-control__control-icon"
                      focusable="false"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.5 6C10.5 6 8 8.5 8 11.5c0 1.1.3 2.1.9 3l-3.4 3 1 1.1 3.4-2.9c1 .9 2.2 1.4 3.6 1.4 3 0 5.5-2.5 5.5-5.5C19 8.5 16.5 6 13.5 6zm0 9.5c-2.2 0-4-1.8-4-4s1.8-4 4-4 4 1.8 4 4-1.8 4-4 4z"
                      />
                    </svg>
                    <div
                      class="components-base-control__field"
                    >
                      <label
                        class="components-base-control__label"
                        for="woocommerce-select-control-0__control-input"
                      >
                        Country / Region *
                      </label>
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Country / Region *"
                        autocomplete="new-password"
                        class="woocommerce-select-control__control-input"
                        id="woocommerce-select-control-0__control-input"
                        placeholder=""
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="components-base-control muriel-component muriel-input-text empty css-wdf2ti-Wrapper ej5x27r4"
                >
                  <div
                    class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                  >
                    <label
                      class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                      for="inspector-text-control-0"
                    >
                      Address
                    </label>
                    <input
                      autocomplete="address-line1"
                      class="components-text-control__input"
                      id="inspector-text-control-0"
                      placeholder="Address"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="components-base-control muriel-component muriel-input-text empty css-wdf2ti-Wrapper ej5x27r4"
                >
                  <div
                    class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                  >
                    <label
                      class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                      for="inspector-text-control-1"
                    >
                      Post code
                    </label>
                    <input
                      autocomplete="postal-code"
                      class="components-text-control__input"
                      id="inspector-text-control-1"
                      placeholder="Post code"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="components-base-control muriel-component muriel-input-text empty css-wdf2ti-Wrapper ej5x27r4"
                >
                  <div
                    class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                  >
                    <label
                      class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                      for="inspector-text-control-2"
                    >
                      City
                    </label>
                    <input
                      autocomplete="address-level2"
                      class="components-text-control__input"
                      id="inspector-text-control-2"
                      placeholder="City"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="components-base-control muriel-component muriel-input-text with-value css-wdf2ti-Wrapper ej5x27r4"
              >
                <div
                  class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                >
                  <label
                    class="components-base-control__label css-eweeby-StyledLabel-labelStyles ej5x27r2"
                    for="inspector-text-control-3"
                  >
                    Email address
                  </label>
                  <input
                    autocomplete="email"
                    checked=""
                    class="components-text-control__input"
                    id="inspector-text-control-3"
                    placeholder="Email address"
                    required=""
                    type="text"
                    value="wordpress@example.com"
                  />
                </div>
              </div>
            </div>
            <div
              class="components-flex-item css-mw3lhz-View-Item-sx-Base em57xhy0"
              data-wp-c16t="true"
              data-wp-component="FlexItem"
            >
              <div
                class="woocommerce-profile-wizard__newsletter-signup"
              >
                <div
                  class="components-base-control components-checkbox-control css-wdf2ti-Wrapper ej5x27r4"
                >
                  <div
                    class="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
                  >
                    <span
                      class="components-checkbox-control__input-container"
                    >
                      <input
                        checked=""
                        class="components-checkbox-control__input"
                        id="inspector-checkbox-control-0"
                        type="checkbox"
                        value="true"
                      />
                      <svg
                        aria-hidden="true"
                        class="components-checkbox-control__checked"
                        focusable="false"
                        height="24"
                        role="presentation"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
                        />
                      </svg>
                    </span>
                    <label
                      class="components-checkbox-control__label"
                      for="inspector-checkbox-control-0"
                    >
                      Get tips, product updates and inspiration straight to your mailbox.
                       
                      <span
                        class="woocommerce-profile-wizard__powered-by-mailchimp"
                      >
                        Powered by Mailchimp
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="components-flex components-card__footer components-card-footer css-142znvl-View-Flex-sx-Base-sx-Items-ItemsRow-Footer-borderRadius-borderColor-medium em57xhy0"
            data-wp-c16t="true"
            data-wp-component="CardFooter"
          >
            <button
              aria-disabled="true"
              class="components-button is-primary"
              disabled=""
              type="button"
            >
              Continue
            </button>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
          data-wp-c16t="true"
          data-wp-component="Elevation"
        />
        <div
          aria-hidden="true"
          class="components-elevation css-91yjwm-View-Elevation-sx-Base-elevationClassName em57xhy0"
          data-wp-c16t="true"
          data-wp-component="Elevation"
        />
      </div>
      <div
        class="woocommerce-profile-wizard__footer"
      >
        <button
          class="components-button woocommerce-profile-wizard__footer-link is-link"
          type="button"
        >
          Skip setup store details
        </button>
        <button
          aria-label="Manual setup is only recommended for
 experienced WooCommerce users or developers."
          class="components-button is-tertiary"
          type="button"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3.2c-4.8 0-8.8 3.9-8.8 8.8 0 4.8 3.9 8.8 8.8 8.8 4.8 0 8.8-3.9 8.8-8.8 0-4.8-4-8.8-8.8-8.8zm0 16c-4 0-7.2-3.3-7.2-7.2C4.8 8 8 4.8 12 4.8s7.2 3.3 7.2 7.2c0 4-3.2 7.2-7.2 7.2zM11 17h2v-6h-2v6zm0-8h2V7h-2v2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
